cdap-app-fabric/src/main/java/io/cdap/cdap/internal/state/AppStateHandler.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.state;
```

Comment:
Should it be in the same package as the other handlers for appfabric (io.cdap.cdap.gateway.handlers) ? Also the binding for this is missing.

Reply:
Moved it to io.cdap.cdap.internal.app because the ApplicationLifecycleService and DefaultStore classes are in this folder.